text-generation-inference/docker/Dockerfile

```diff
@@ -119,10 +119,16 @@ COPY --from=builder /usr/src/target/release/text-generation-launcher /usr/local/
 COPY --from=pyserver /pyserver/build/dist dist
 RUN pip install dist/text_generation_server*.tar.gz
 
-# TPU compatible image
-FROM tpu_base
 
-COPY text-generation-inference/entrypoint.sh entrypoint.sh
+# TPU compatible image for Inference Endpoints
+FROM tpu-base as inference-endpoint
+
+COPY text-generation-inference/docker/entrypoint.sh entrypoint.sh
 RUN chmod +x entrypoint.sh
 
-ENTRYPOINT ["./entrypoint.sh"]
\ No newline at end of file
+ENTRYPOINT ["./entrypoint.sh"]
+
+# TPU compatible image
+FROM tpu_base as final_image
+ENTRYPOINT text-generation-launcher
+CMD ["--json-output"]
```

Comment:
this seems to revert https://github.com/huggingface/optimum-tpu/pull/28/commits/8756c8a1753eb538b9069dc88b1fe001403ef68d, are you sure about it?

Reply:
For the case of non inference-endpoints (IE) Docker image we want the image to provide their own argument, so this was correct and the changes we did were too focus on IE loosing flexibility in other deployment scenarios

Reply:
Ok, in that case we might want to keep 2 docker images, one for TGI non inference-endpoints (IE), the other for IE. Or, if we are not interested in maintaining one of those, we should probably drop the image and the associated targets.

Reply:
Yes but we are definitely interested in keeping both and potentially adding a third one for vertex if it makes sense later 👍🏻